packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -3205,3 +3207,36 @@ function finishPendingInteractions(root, committedExpirationTime) {
     );
   }
 }
+
+function detachFiberPair(fiber: Fiber) {
+  if (fiber.effectTag & Deletion) {
```

Comment:
Does this function get inlined? If not, I wonder if we should put the conditional on the outside. My theory is that we save a bit because the inline cache type check is already done against the fiber at the parent callsite so it'll be quicker than to do another type check here. It doesn't matter if it's a Deletion but Deletions are only a fraction of all effects. So the common case is that this doesn't happen and in that case we can skip the function call and additional hidden class type check.